plugins/hls-selection-range-plugin/src/Ide/Plugin/SelectionRange.hs

```diff
@@ -1,67 +1,66 @@
+                                                       uriToFilePath')
+import           Development.IDE.Core.Actions         (useE)
+import           Development.IDE.Core.PositionMapping (PositionMapping,
+                                                       fromCurrentPosition,
+                                                       toCurrentRange)
+import           Development.IDE.Types.Logger         (Pretty (..))
+import           Ide.Plugin.SelectionRange.CodeRange  (CodeRange (..),
+                                                       GetCodeRange (..),
+                                                       codeRangeRule)
+import qualified Ide.Plugin.SelectionRange.CodeRange  as CodeRange
+import           Ide.PluginUtils                      (pluginResponse,
+                                                       positionInRange)
+import           Ide.Types                            (PluginDescriptor (pluginHandlers, pluginRules),
+                                                       PluginId,
+                                                       defaultPluginDescriptor,
+                                                       mkPluginHandler)
+import           Language.LSP.Server                  (LspM)
+import           Language.LSP.Types                   (List (List),
+                                                       NormalizedFilePath,
+                                                       Position (..),
+                                                       ResponseError,
+                                                       SMethod (STextDocumentSelectionRange),
+                                                       SelectionRange (..),
+                                                       SelectionRangeParams (..),
+                                                       TextDocumentIdentifier (TextDocumentIdentifier),
+                                                       Uri)
+import           Prelude                              hiding (log, span)
 
-descriptor :: PluginId -> PluginDescriptor IdeState
-descriptor plId = (defaultPluginDescriptor plId)
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId = (defaultPluginDescriptor plId)
     { pluginHandlers = mkPluginHandler STextDocumentSelectionRange selectionRangeHandler
+    -- TODO @sloorush add folding range
+    -- <> mkPluginHandler STextDocumentFoldingRange foldingRangeHandler
+    , pluginRules = codeRangeRule (cmapWithPrio LogCodeRange recorder)
     }
 
+data Log = LogCodeRange CodeRange.Log
+
+instance Pretty Log where
+    pretty log = case log of
+        LogCodeRange codeRangeLog -> pretty codeRangeLog
+
 selectionRangeHandler :: IdeState -> PluginId -> SelectionRangeParams -> LspM c (Either ResponseError (List SelectionRange))
 selectionRangeHandler ide _ SelectionRangeParams{..} = do
-    liftIO $ logDebug logger $ "requesting selection range for file: " <> T.pack (show uri)
```

Comment:
did you decide this wasn't worth it or you just don't want to pass the recorder here?

Reply:
I believe this information should be logged systematically for all handlers if needed. Currently, I do not need it anymore.